File: language-testing/fibonacci/fibonacci_untested.rb
#! /usr/bin/env ruby

def fib(rot)
  old_value = 1
  new_value = 0

  while rot > 0
    old_value, new_value = new_value, new_value + old_value
    rot -= 1
  end

  new_value
end

# If no arguments; run 20 fibs, else; fib the first argument:
if ARGV.length == 0 then (0...10).each { |i| puts "#{fib(i)}\t#{fib(i + 10)}" }
else                     puts fib(ARGV[0].to_i)
end

# Name: Fibonacci Sequence
# Language: Ruby
# State: Done
#
# Prints out numbers from the fibonacci sequence
#
#
# Example: ./fibonacci.rb
# Example2: ./fibonacci.rb 42

File: language-testing/fibonacci/fibonacci_untested.pl
#! /usr/bin/env perl

use 5.14.2;
use strict;
use warnings;

sub fib {

    my $oldval = 1;
    my $newval = 0;
    my $tmp = 0;

    while (--$_[0] >= 0) {
        $tmp = $oldval;
        $oldval = $newval;
        $newval = $tmp + $oldval;
    }
    return $newval;
}

# Fib the arugment, if possible:
if ($#ARGV == 0) {
    printf "%d\n", &fib($ARGV[0]);
    exit 0;
}
# Otherwise, run 20 fibs
for (0..9) {
    printf "%d\t%d\n", &fib($_+0), &fib($_+10);
}

=pod TAIL INFO
Name: Fibonacci Sequence
Language: Perl
State: Done
Created: 2013-08-08

Prints out numbers from the fibonacci sequence

Example1: ./fibonacci.pl
Example2: ./fibonacci.pl 42
=cut
